src/App.tsx

```diff
@@ -15,71 +16,99 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+type State = {
+  selectedGood: string;
+};
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+export class App extends React.Component<{}, State> {
+  state = {
+    selectedGood: 'Jam',
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  clearHandler = () => {
+    this.setState({
+      selectedGood: '',
+    });
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
+  addHandler = (good: string) => {
+    this.setState({
+      selectedGood: good,
+    });
+  };
+
+  render() {
+    const { selectedGood } = this.state;
+
+    return (
+      <main className="main section container">
+        {!selectedGood ? (
+          <h1 className="title">No goods selected</h1>
+        ) : (
+          <h1 className="title is-flex is-align-items-center">
+            {`${selectedGood} is selected`}
+
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
             <button
-              data-cy="AddButton"
+              data-cy="ClearButton"
               type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+              className="delete ml-3"
+              onClick={this.clearHandler}
+            />
+          </h1>
+        )}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+        <table className="table">
+          <tbody>
+            {goods.map(good => (
+              <tr
+                data-cy="Good"
+                className={
+                  cn({
+                    'has-background-success-light': good === selectedGood,
+                  })
+                }
+              >
+                <td>
+                  {good === selectedGood ? (
+                    <button
+                      data-cy="RemoveButton"
+                      type="button"
+                      className="button is-info"
+                      onClick={this.clearHandler}
+                    >
+                      -
+                    </button>
+                  ) : (
+                    <button
+                      data-cy="AddButton"
+                      type="button"
+                      className="button"
+                      onClick={() => {
+                        this.addHandler(good);
+                      }}
+                    >
+                      +
+                    </button>
```

Comment:
it might be tidier to assign remove/add button to variables
```
const removeButton = (
<button
  data-cy="RemoveButton"
  type="button"
  className="button is-info"
  onClick={this.clearHandler}
>
  -
</button>
)

const addButton = ...
```

```suggestion
                <td>
                  {good === selectedGood
                    ? removeButton
                    : addButton
                  }
```